frontend/public/components/secrets/create-secret/utils.ts

```diff
@@ -94,3 +97,38 @@ export const useSecretDescription = (typeAbstraction: SecretTypeAbstraction): st
       return null;
   }
 };
+
+export const opaqueEntriesToObject = (opaqueEntriesArray: OpaqueDataEntry[]): SecretStringData => {
+  return _.reduce(
+    opaqueEntriesArray,
+    (acc, entry) =>
+      _.assign(acc, {
+        [entry?.key]: entry?.isBase64 || entry?.isBinary ? entry.value : Base64.encode(entry.value),
+      }),
+    {},
+  );
+};
```

Comment:
nit, simplify and prefer native JS over lodash:
```suggestion
export const opaqueEntriesToObject = (opaqueEntriesArray: OpaqueDataEntry[]): SecretStringData =>
  (opaqueEntriesArray ?? []).reduce(
    (acc, entry) => ({
      ...acc,
      [entry.key]: entry.isBinary ? entry.value : Base64.encode(entry.value),
    }),
    {},
  );
```